### ISSUE TITLE
stack build copies unnecessary executables when asked to build just a single executable

### ISSUE THREAD

**Issue by athas** [commercialhaskell/stack#3486]
### Steps to reproduce

1. Have a repository with several executable targets.  Say, the project is `foo`, and the executables are `exe1`, `exe2`, `exe3`.
2. Run `stack build foo:exe3`

### Expected

Stack should build the `exe3` target (and any dirty dependencies) and then *only* copy the `exe3` binary from the `.stack-work/dist/.../build/` directory to `.stack-work/install/.../bin/`.

### Actual

Stack copies *all* executables in the package, even the ones with no changes and which it was not asked to copy, which is slow if there are a lot of them.

### Stack version

```
$ stack --version
Version 1.5.1, Git revision 96e83a05fbeafb5cc3b1f52195433101c1f6ab8a x86_64 hpack-0.18.1
```

### Method of installation

Via `stack upgrade`.

**Comment by mgsloan** [comment]
This is a symptom of https://github.com/commercialhaskell/stack/issues/1406 , which at its root is a cabal issue.  It seemed like this was going to get fixed in cabal, there was even a PR, but it got left out of releases for some reason.

Worth revisiting.  For now, you'll have to live with it.  One option is to add cabal flags to set `buildable: false` on particular executables.

**Comment by 23Skidoo** [comment]
> This is a symptom of #1406 , which at its root is a cabal issue. It seemed like this was going to get fixed in cabal, there was even a PR, but it got left out of releases for some reason.

Can someone please summarise the issue and post a ticket to the Cabal bug tracker (or post a comment in an existing thread, if there's already a ticket)?

**Comment by mgsloan** [comment]
@23Skidoo Sorry for the delay on following up on this.  I've opened https://github.com/commercialhaskell/stack/issues/3653 to track looking into whether this is resolved in Cabal-2.0, and changing how things work if it does.  If not, an issue will be opened to track it.

**Comment by mgsloan** [comment]
Based on ezyang's comment in  https://github.com/haskell/cabal/issues/2780#issuecomment-354490838 , it should be feasible to move forward on this at least when Cabal >= 2.0
